File: impl/src/org/ufomatics/archipelago/view/swing/TypePane.java
package org.ufomatics.archipelago.view.swing;

import org.ufomatics.archipelago.control.UFOPlatformController;
import org.ufomatics.archipelago.model.MetaType;

import javax.swing.*;
import java.util.Set;

public class TypePane extends JPanel {
    /**
     * The data. Will be modified if ok, or not if cancelled
     */
    private MetaType data;

    private JTextField nameTextfield;
    private JTextArea descriptionTextarea;
    private JTextField ancestorsTextfield;
    private JButton button1;
    private Set ancestors;

    private UFOPlatformController controller;

    public TypePane(UFOPlatformController controller, MetaType data) {
        this.controller = controller;
        this.data = data;

        SpringLayout layout = new SpringLayout();
        setLayout(layout);

        JLabel nameLabel = new JLabel("Name");
        nameTextfield = new JTextField(data.getName());
        add(nameLabel);
        add(nameTextfield);
        layout.putConstraint(SpringLayout.EAST, this, 5, SpringLayout.EAST, nameTextfield);
        layout.putConstraint(SpringLayout.SOUTH, this, 5, SpringLayout.SOUTH, nameTextfield);
        layout.putConstraint(SpringLayout.WEST, nameLabel, 5, SpringLayout.WEST, this);
        layout.putConstraint(SpringLayout.NORTH, nameLabel, 5, SpringLayout.NORTH, this);
        layout.putConstraint(SpringLayout.WEST, nameTextfield, 5, SpringLayout.EAST, nameLabel);
        layout.putConstraint(SpringLayout.NORTH, nameTextfield, 0, SpringLayout.NORTH, nameLabel);
    }

    public boolean requestFocusInWindow() {
        return nameTextfield.requestFocusInWindow();
    }

    public void ok() {
        data.setName(nameTextfield.getText());
    }
}

File: src/archipelago/impl/view/ui/TypePane.ts
import {UFOPlatformController} from "archipelago/api/control/UFOPlatformController"
import {MetaType} from "archipelago/api/model/MetaType"
import {JLabel, JPanel, JTextField, SpringLayout} from "ts-jsdk"
import {SpringLayoutDirection} from "ts-jsdk/dist/awt/SpringLayout"

export class TypePane extends JPanel {

  private readonly nameTextfield: JTextField

  /* private descriptionTextarea: JTextArea
   private ancestorsTextfield: JTextField
   private button1: JButton
   private ancestors = new Set()*/

  constructor(
    protected controller: UFOPlatformController,
    /**
     * The data. Will be modified if ok, or not if cancelled
     */
    protected data: MetaType
  ) {
    super()
    const layout = new SpringLayout()
    this.setLayout(layout)

    const nameLabel = new JLabel("Name")
    this.nameTextfield = new JTextField(data.getName())
    this.add(nameLabel)
    this.add(this.nameTextfield)
    layout.putConstraint(SpringLayoutDirection.EAST, this, 5, SpringLayoutDirection.EAST, this.nameTextfield)
    layout.putConstraint(SpringLayoutDirection.SOUTH, this, 5, SpringLayoutDirection.SOUTH, this.nameTextfield)
    layout.putConstraint(SpringLayoutDirection.WEST, nameLabel, 5, SpringLayoutDirection.WEST, this)
    layout.putConstraint(SpringLayoutDirection.NORTH, nameLabel, 5, SpringLayoutDirection.NORTH, this)
    layout.putConstraint(SpringLayoutDirection.WEST, this.nameTextfield, 5, SpringLayoutDirection.EAST, nameLabel)
    layout.putConstraint(SpringLayoutDirection.NORTH, this.nameTextfield, 0, SpringLayoutDirection.NORTH, nameLabel)
  }

  requestFocusInWindow(): boolean {
    return this.nameTextfield.requestFocusInWindow()
  }

  ok(): void {
    this.data.setName(this.nameTextfield.getText())
  }
}
